### PR TITLE
BEAM-10221: Add in example tests cases of Kotlin cookbook

### PR DESCRIPTION
**Pull request by perkss**
* Add in four test cases based on the Java equivalent to the Kotlin examples module. 
* Shows how to write the tests that were missing for the examples.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [x] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Samza | Spark
--- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Java11/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Java11/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python2_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)
XLang | --- | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Spark/lastCompletedBuild/)

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by perkss** [comment]
@harshithdwivedi please can you check a look

File: examples/kotlin/src/test/java/org/apache/beam/examples/kotlin/cookbook/JoinExamplesTest.kt [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+                KV.of("VM", "Date: 20141212, Actor1: LAOS, url: http://www.chicagotribune.com"),
+                KV.of("BE", "Date: 20141213, Actor1: AFGHANISTAN, url: http://cnn.com"),
+                KV.of("VM", "Date: 20141212, Actor1: BANGKOK, url: http://cnn.com"))
+        private val PARSED_COUNTRY_CODES = listOf(KV.of("BE", "Belgium"), KV.of("VM", "Vietnam"))
+        private val cc1 = TableRow().set("FIPSCC", "VM").set("HumanName", "Vietnam")
+        private val cc2 = TableRow().set("FIPSCC", "BE").set("HumanName", "Belgium")
+        private val CCS = arrayOf(cc1, cc2)
+        val CC_ARRAY = listOf(*CCS)
+        val JOINED_EVENTS = arrayOf("Country code: VM, Country name: Vietnam, Event info: Date: 20141212, Actor1: LAOS, "
+                + "url: http://www.chicagotribune.com", "Country code: VM, Country name: Vietnam, Event info: Date: 20141212, Actor1: BANGKOK, "
```
Comment (harshithdwivedi) [inline examples/kotlin/src/test/java/org/apache/beam/examples/kotlin/cookbook/JoinExamplesTest.kt:68]: Can we get rid of this rogue `"`?

Comment (perkss) [reply examples/kotlin/src/test/java/org/apache/beam/examples/kotlin/cookbook/JoinExamplesTest.kt]: Thanks for looking through. Not sure what you mean here specifically but formatted inline with max width now and how the java example was shown

**Comment by perkss** [comment]
@pabloem could you please take a look?

**Comment by pabloem** [comment]
retest this please

**Comment by pabloem** [comment]
RAT Precommit - can you please add an Apache License header to any new files? (you can copy the header from other kotlin files)

**Comment by perkss** [comment]
retest this please

**Comment by pabloem** [comment]
retest this please

**Comment by pabloem** [comment]
retest this please

**Comment by perkss** [comment]
Run Java PreCommit

**Comment by perkss** [comment]
retest this please

**Comment by perkss** [comment]
retest this please

**Comment by perkss** [comment]
retest this please

**Comment by pabloem** [comment]
Run Java PreCommit

**Comment by pabloem** [comment]
retest this please

**Comment by perkss** [comment]
Run Java PreCommit

**Comment by perkss** [comment]
retest this please

**Comment by perkss** [comment]
They do actually seem to be running fine [here](https://builds.apache.org/job/beam_PreCommit_Java_Commit/lastCompletedBuild/testReport/org.apache.beam.examples.kotlin.cookbook/)

**Comment by pabloem** [comment]
great! thanks for pointhing that out

**Comment by pabloem** [comment]
let me get a passing run for the tests and we'll merge

**Comment by perkss** [comment]
@pabloem thanks.

**Comment by pabloem** [comment]
Run Java PreCommit

**Comment by perkss** [comment]
Run Java PreCommit

**Comment by perkss** [comment]
@pabloem seems like it got stuck on something else and got aborted https://builds.apache.org/job/beam_PreCommit_Java_Commit/11770/

**Comment by pabloem** [comment]
Run Java PreCommit

**Comment by pabloem** [comment]
that's right @perkss it looks like an unrelated issue, so I'll just run 'till I get a passing run and merge

**Comment by perkss** [comment]
Thanks @pabloem for you perseverance all green now.

**Comment by pabloem** [comment]
thanks @perkss! I'm quite happy to get tests on the examples!

**Pull request merged**